test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReopenCaseControllerSpec.scala

```diff
@@ -0,0 +1,179 @@
+    }
+
+  }
+
+  "Confirm Refer a Case" should {
+
+    "return OK and HTML content type" in {
+      when(casesService.getOne(refEq("reference"))(any[HeaderCarrier])).thenReturn(successful(Some(caseWithStatusREFERRED)))
+      when(casesService.reopenCase(refEq(caseWithStatusREFERRED), refEq(operator))(any[HeaderCarrier])).thenReturn(successful(caseWithStatusOPEN))
+
+      val result: Result = await(controller.confirmReopenCase("reference")(newFakePOSTRequestWithCSRF()))
+
+      status(result) shouldBe Status.OK
+      contentTypeOf(result) shouldBe Some(MimeTypes.HTML)
+      charsetOf(result) shouldBe Some("utf-8")
+      bodyOf(result) should include("This case has been reopen")
+    }
+
+
+    def doNotRedirectWith(allowedStatus: CaseStatus*) = {
+
+      for (s <- CaseStatus.values) {
+        val statusCase = Cases.btiCaseExample.copy(status = s)
+        when(casesService.getOne(refEq("reference"))(any[HeaderCarrier])).thenReturn(successful(Some(statusCase)))
+        when(casesService.reopenCase(any[Case], refEq(operator))(any[HeaderCarrier])).thenReturn(successful(statusCase))
+        val result: Result = await(controller.confirmReopenCase("reference")(newFakePOSTRequestWithCSRF()))
+        if (allowedStatus.contains(s)) {
+          withClue(s"Status $s must be redirected") {
+            status(result) shouldBe Status.OK
+          }
+        } else {
+          withClue(s"Status $s has not been redirected") {
+            status(result) shouldBe Status.SEE_OTHER
+          }
+          contentTypeOf(result) shouldBe None
+          charsetOf(result) shouldBe None
+          locationOf(result) shouldBe Some("/tariff-classification/cases/reference/application")
+        }
+      }
+    }
+
+
+    "redirect to Application Details for non REFERRED or SUSPENDED statuses" in {
+      doNotRedirectWith(CaseStatus.REFERRED, CaseStatus.SUSPENDED)
+    }
+
+    "return Not Found and HTML content type on missing Case" in {
+      when(casesService.getOne(refEq("reference"))(any[HeaderCarrier])).thenReturn(successful(None))
+      val result: Result = await(controller.confirmReopenCase("reference")(newFakePOSTRequestWithCSRF()))
+
+      status(result) shouldBe Status.OK
+      contentTypeOf(result) shouldBe Some(MimeTypes.HTML)
+      charsetOf(result) shouldBe Some("utf-8")
+      bodyOf(result) should include("We could not find a Case with reference")
+    }
+
+    "propagate the error in case the CaseService fails to release the case" in {
+      val error = new IllegalStateException("expected error")
+
+      when(casesService.getOne(refEq("reference"))(any[HeaderCarrier])).thenReturn(failed(error))
+
+      val caught = intercept[error.type] {
+        await(controller.confirmReopenCase("reference")(newFakePOSTRequestWithCSRF()))
+      }
+      caught shouldBe error
+    }
+  }
+
+  private def newFakeGETRequestWithCSRF(): FakeRequest[AnyContentAsEmpty.type] = {
```

Comment:
These private methods are in few test classes
I would extract them as protected methods in a trait to be re-used in all the tests that use them